File: app/src/main/java/com/karanchuk/roman/testtranslate/data/network/YandexDictionaryApi.java
package com.karanchuk.roman.testtranslate.data.network;

import com.karanchuk.roman.testtranslate.data.database.model.DictDefinition;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by roman on 17.6.17.
 */

public interface YandexDictionaryApi {

    @GET("api/v1/dicservice.json/lookup?")
    Single<DictDefinition> getValueFromDictionary(@Query("key") String key,
                                                  @Query("text") String text,
                                                  @Query("lang") String lang);
}

File: app/src/main/java/com/romankaranchuk/translator/data/network/YandexDictionaryApi.kt
package com.romankaranchuk.translator.data.network

import com.romankaranchuk.translator.data.database.model.DictDefinition
import retrofit2.http.GET
import retrofit2.http.Query

interface YandexDictionaryApi {

    @GET("api/v1/dicservice.json/lookup?")
    suspend fun getDictDefinition(
        @Query("key") key: String,
        @Query("text") text: String,
        @Query("lang") lang: String
    ): DictDefinition?
}